enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/atomicbroadcast/multipaxos/PaxosInstance.java

```diff
@@ -191,10 +199,43 @@ public PaxosInstance snapshot(PaxosInstanceStore store)
     @Override
     public String toString()
     {
-        return "[id:" + id +
-               " state:" + state.name() +
-               " b:" + ballot +
-               " v1:" + value_1 +
-               " v2:" + value_2 + "]";
+        try
+        {
+            Object toStringValue1 = null;
+            if (value_1 != null)
+            {
+                if ( value_1 instanceof Payload )
+                {
+                    toStringValue1 = new AtomicBroadcastSerializer().receive( (Payload) value_1 ).toString();
+                }
+                else
+                {
+                    toStringValue1 = value_1.toString();
+                }
+            }
+
+            Object toStringValue2 = null;
+            if (value_2 != null)
+            {
+                if ( value_2 instanceof Payload )
+                {
+                    toStringValue2 = new AtomicBroadcastSerializer().receive( (Payload) value_2 ).toString();
+                }
+                else
+                {
+                    toStringValue2 = value_2.toString();
+                }
+            }
+
+            return "[id:" + id +
+                   " state:" + state.name() +
+                   " b:" + ballot +
+                   " v1:" + toStringValue1 +
+                   " v2:" + toStringValue2 + "]";
+        }
+        catch ( Throwable e )
+        {
+            return "";
```

Comment:
can't we do better than this? perhaps skipping only the values causing the issues?